homeassistant/components/motion_blinds/cover.py

```diff
@@ -288,6 +311,12 @@ async def async_set_cover_position(self, **kwargs):
             await self.hass.async_add_executor_job(
                 self._blind.Set_position, 100 - position
             )
+        if position > self._blind.position:
+            self._closing = True
+            self._opening = False
+        else:
+            self._opening = True
+            self._closing = False
```

Comment:
This could be a function since its duplicated in `async_set_absolute_position`

Reply:
There are other places as well that could be broken out

Reply:
I made some simplifications to reduce the amount of code.